In [1]:
import pandas as pd
import requests
from bs4 import BeautifulSoup
import numpy as np

In [2]:
requests.get('https://www.ambitionbox.com/list-of-companies?page=1')

<Response [403]>

## if response code is 403

#### use - headers={'User-Agent':'Mozilla/5.0 (Windows NT 6.3; Win 64 ; x64) Apple WeKit /537.36(KHTML , like Gecko) Chrome/80.0.3987.162 Safari/537.36'}

In [3]:
url='https://www.ambitionbox.com/list-of-companies?page=1'
headers={'User-Agent':'Mozilla/5.0 (Windows NT 6.3; Win 64 ; x64) Apple WeKit /537.36(KHTML , like Gecko) Chrome/80.0.3987.162 Safari/537.3'}
webpage=requests.get(url, headers=headers).text
    
    

In [5]:
#webpage

In [6]:
soup=BeautifulSoup(webpage,'lxml')

In [8]:
#print(soup.prettify()) -- to see HTML code in proper format

### CONSIDERING THE WHOLE CONTAINER WHERE ALL THE DETAILS PRESENT FOR COMPANY

In [112]:
company = soup.find_all('div',class_='companyCardWrapper')
len(company)

20

In [113]:
company

[<div class="companyCardWrapper" itemprop="itemListElement" itemscope="itemscope" itemtype="http://schema.org/ListItem"><meta content="1" itemprop="position"/> <meta content="TCS" itemprop="name"/> <meta content="Tata Consultancy Services" itemprop="alternateName"/> <meta content="https://www.ambitionbox.com/overview/tcs-overview" itemprop="url"/> <meta content="https://static.ambitionbox.com/alpha/company/photos/logos/tcs.jpg" itemprop="image"/> <div class="companyCardWrapper__primaryInformation"><div class="companyCardWrapper__companyDetails"><div class="companyCardWrapper__companyLogo"><img alt="Tata Consultancy Services logo" height="50" loading="lazy" onerror="this.onerror=null;this.src='/static/icons/company-placeholder.svg';" src="https://static.ambitionbox.com/assets/v2/images/rs:fit:200:200:false:false/bG9jYWw6Ly8vbG9nb3Mvb3JpZ2luYWxzL3Rjcy5qcGc.webp" width="50"/></div> <div class="companyCardWrapper__metaInformation"><div class="companyCardWrapper__header"><div class="company

In [133]:
name=[]
ratings=[]
reviews=[]
other_info=[]
for i in company:
    name.append(i.find('h2').text.strip())
    ratings.append(i.find('span', class_='companyCardWrapper__companyRatingValue').text.strip())
    reviews.append(i.find('span', class_='companyCardWrapper__ActionCount').text.strip())
    other_info.append(i.find('span',class_='companyCardWrapper__interLinking').text.strip())

    
d={'CompanyName':name,'Ratings':ratings,'ReviewsCounts':reviews,'OtherInfo':other_info}
df=pd.DataFrame(d)
df[['IndustryType','Employees','CompanyType','CompanyAge','HQ & Other Locations']]=df['OtherInfo'].str.split(' \| ', expand=True)
df=df[['CompanyName', 'Ratings', 'ReviewsCounts', 'IndustryType','Employees','OtherInfo']]

In [135]:
df

CompanyName Ratings ReviewsCounts                IndustryType  \
0                       TCS     3.8         74.9k    IT Services & Consulting   
1                 Accenture     4.0         46.7k    IT Services & Consulting   
2                 Cognizant     3.9         41.9k    IT Services & Consulting   
3                     Wipro     3.8         40.3k    IT Services & Consulting   
4                 Capgemini     3.8         34.6k    IT Services & Consulting   
5                 HDFC Bank     4.0         33.7k                     Banking   
6                ICICI Bank     4.0         33.4k                     Banking   
7                   Infosys     3.8         32.5k    IT Services & Consulting   
8                   HCLTech     3.7         29.3k    IT Services & Consulting   
9             Tech Mahindra     3.7         27.7k    IT Services & Consulting   
10                  Genpact     3.9         26.3k    IT Services & Consulting   
11                Axis Bank     3.9           22k                     Banking   
12          Teleperformance     3.8         22.9k                         BPO   
13   Concentrix Corporation     3.9         21.6k                         BPO   
14                      Jio     4.0         20.3k                     Telecom   
15                   Amazon     4.1         21.1k                    Internet   
16                      IBM     4.1         19.4k    IT Services & Consulting   
17          Reliance Retail     4.0         19.2k                      Retail   
18   HDB Financial Services     4.0         17.7k                        NBFC   
19  Larsen & Toubro Limited     4.1           16k  Engineering & Construction   

               Employees                                          OtherInfo  
0      1 Lakh+ Employees  IT Services & Consulting | 1 Lakh+ Employees |...  
1      1 Lakh+ Employees  IT Services & Consulting | 1 Lakh+ Employees |...  
2      1 Lakh+ Employees  IT Services & Consulting | 1 Lakh+ Employees |...  
3      1 Lakh+ Employees  IT Services & Consulting | 1 Lakh+ Employees |...  
4      1 Lakh+ Employees  IT Services & Consulting | 1 Lakh+ Employees |...  
5      1 Lakh+ Employees  Banking | 1 Lakh+ Employees | Public | 30 year...  
6      1 Lakh+ Employees  Banking | 1 Lakh+ Employees | Public | 30 year...  
7      1 Lakh+ Employees  IT Services & Consulting | 1 Lakh+ Employees |...  
8      1 Lakh+ Employees  IT Services & Consulting | 1 Lakh+ Employees |...  
9      1 Lakh+ Employees  IT Services & Consulting | 1 Lakh+ Employees |...  
10     1 Lakh+ Employees  IT Services & Consulting | 1 Lakh+ Employees |...  
11  50k-1 Lakh Employees  Banking | 50k-1 Lakh Employees | Public | 31 y...  
12  50k-1 Lakh Employees  BPO | 50k-1 Lakh Employees | Forbes Global 200...  
13  50k-1 Lakh Employees  BPO | 50k-1 Lakh Employees | Public | 41 years...  
14  50k-1 Lakh Employees  Telecom | 50k-1 Lakh Employees | Public | 6 ye...  
15     1 Lakh+ Employees  Internet | 1 Lakh+ Employees | Public | 30 yea...  
16     1 Lakh+ Employees  IT Services & Consulting | 1 Lakh+ Employees |...  
17     1 Lakh+ Employees  Retail | 1 Lakh+ Employees | 18 years old | Na...  
18     1 Lakh+ Employees  NBFC | 1 Lakh+ Employees | 17 years old | Ahme...  
19     10k-50k Employees  Engineering & Construction | 10k-50k Employees...

# Final Code for changing pages and scrape data

In [162]:
import requests
from bs4 import BeautifulSoup
import pandas as pd

# Initialize an empty DataFrame to store all data
final = pd.DataFrame()

headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 6.3; Win 64 ; x64) Apple WeKit /537.36(KHTML , like Gecko) Chrome/80.0.3987.162 Safari/537.3'}

# Loop through the pages
for j in range(1, 4):  # Scraping only limited pages
    url = 'https://www.ambitionbox.com/list-of-companies?page={}'.format(j)
    webpage = requests.get(url, headers=headers).text
    soup = BeautifulSoup(webpage, 'lxml')
    
    company = soup.find_all('div', class_='companyCardWrapper')
    
    name = []
    ratings = []
    reviews = []
    other_info = []
    
    for i in company:
        name.append(i.find('h2').text.strip())
        ratings.append(i.find('span', class_='companyCardWrapper__companyRatingValue').text.strip())
        reviews.append(i.find('span', class_='companyCardWrapper__ActionCount').text.strip())
        other_info.append(i.find('span', class_='companyCardWrapper__interLinking').text.strip())
    
    d = {'CompanyName': name, 'Ratings': ratings, 'ReviewsCounts': reviews, 'OtherInfo': other_info}
    
    # Create a DataFrame for the current page
    df = pd.DataFrame(d)
    df[['IndustryType', 'Employees', 'CompanyType', 'CompanyAge', 'HQ & Other Locations']] = df['OtherInfo'].str.split(' \| ', expand=True)
    df = df[['CompanyName', 'Ratings', 'ReviewsCounts', 'IndustryType', 'Employees', 'CompanyType', 'CompanyAge', 'HQ & Other Locations']]
    
    # Append the current page data to the final DataFrame
    final = pd.concat([final, df], ignore_index=True)
    

# Display the final DataFrame

final=final[['CompanyName', 'Ratings', 'ReviewsCounts', 'IndustryType', 'Employees']]
#print(final)

In [163]:
final

CompanyName Ratings ReviewsCounts  \
0                          TCS     3.8         74.9k   
1                    Accenture     4.0         46.7k   
2                    Cognizant     3.9         41.9k   
3                        Wipro     3.8         40.3k   
4                    Capgemini     3.8         34.6k   
5                    HDFC Bank     4.0         33.7k   
6                   ICICI Bank     4.0         33.4k   
7                      Infosys     3.8         32.5k   
8                      HCLTech     3.7         29.3k   
9                Tech Mahindra     3.7         27.7k   
10                     Genpact     3.9         26.3k   
11                   Axis Bank     3.9           22k   
12             Teleperformance     3.8         22.9k   
13      Concentrix Corporation     3.9         21.6k   
14                         Jio     4.0         20.3k   
15                      Amazon     4.1         21.1k   
16                         IBM     4.1         19.4k   
17             Reliance Retail     4.0         19.2k   
18      HDB Financial Services     4.0         17.7k   
19     Larsen & Toubro Limited     4.1           16k   
20               Vodafone Idea     4.2         14.1k   
21                      BYJU'S     3.1         15.2k   
22         Kotak Mahindra Bank     3.8         14.2k   
23                    Deloitte     3.9         14.2k   
24         Reliance Industries     4.1         13.4k   
25                  iEnergizer     4.7         15.5k   
26               Bharti Airtel     4.1         11.8k   
27                 Tata Motors     4.2         10.2k   
28                         WNS     3.5          9.6k   
29                    Flipkart     4.1          9.4k   
30               IndusInd Bank     3.7          9.1k   
31               Ernst & Young     3.6          9.1k   
32       AU Small Finance Bank     4.4          8.8k   
33         Mahindra & Mahindra     4.2          8.4k   
34                 Infosys BPM     3.8          8.3k   
35              DXC Technology     3.8          8.3k   
36             IDFC FIRST Bank     4.0          8.5k   
37  Conneqt Business Solutions     3.7          7.9k   
38                 LTIMindtree     3.6          8.2k   
39                   HDFC Life     4.1          7.7k   
40               Bajaj Finserv     4.1          7.7k   
41                     Mphasis     3.5            7k   
42                         PwC     3.6          7.2k   
43                  HSBC Group     4.1          6.7k   
44                     Startek     3.3          6.6k   
45    Hinduja Global Solutions     4.0          6.4k   
46                     Samsung     4.1          6.3k   
47                    Ericsson     4.3          6.3k   
48               Maruti Suzuki     4.2          6.5k   
49             Shriram Finance     4.1          6.3k   
50                    Yes Bank     3.9          6.3k   
51                Asian Paints     4.1          6.3k   
52                       Cipla     4.1            6k   
53                 Dr. Reddy's     4.2          5.9k   
54  Sutherland Global Services     3.7          5.8k   
55                       Paytm     3.4          6.1k   
56                Bandhan Bank     3.8          6.1k   
57            UltraTech Cement     4.2          5.8k   
58                 EXL Service     3.8            6k   
59                    NTT DATA     4.1          5.8k   

                         IndustryType             Employees  
0            IT Services & Consulting     1 Lakh+ Employees  
1            IT Services & Consulting     1 Lakh+ Employees  
2            IT Services & Consulting     1 Lakh+ Employees  
3            IT Services & Consulting     1 Lakh+ Employees  
4            IT Services & Consulting     1 Lakh+ Employees  
5                             Banking     1 Lakh+ Employees  
6                             Banking     1 Lakh+ Employees  
7            IT Services & Consulting     1 Lakh+ Employees  
8            IT Services & Consulting     1 